plugins/gnostic-linter/main.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/protobuf/proto"
+	lint "github.com/googleapis/gnostic/metrics/lint"
+	openapiv2 "github.com/googleapis/gnostic/openapiv2"
+	openapiv3 "github.com/googleapis/gnostic/openapiv3"
+	plugins "github.com/googleapis/gnostic/plugins"
+)
+
+// Record an error, then serialize and return a response.
+func sendAndExitIfError(err error, response *plugins.Response) {
+	if err != nil {
+		response.Errors = append(response.Errors, err.Error())
+		sendAndExit(response)
+	}
+}
+
+// Serialize and return a response.
+func sendAndExit(response *plugins.Response) {
+	responseBytes, _ := proto.Marshal(response)
+	os.Stdout.Write(responseBytes)
+	os.Exit(0)
+}
+
+// This is the main function for the plugin.
+func main() {
+	env, err := plugins.NewEnvironment()
+	env.RespondAndExitIfError(err)
+
+	var linter *lint.Linter
+	f, err := os.OpenFile("results.txt",
+		os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Println(err)
+	}
+	defer f.Close()
+
+	for _, model := range env.Request.Models {
+		switch model.TypeUrl {
+		case "openapi.v2.Document":
+			documentv2 := &openapiv2.Document{}
+			err = proto.Unmarshal(model.Value, documentv2)
+			if err == nil {
+				// Analyze the API document.
+				linter, _ = lint.AIPLinterV2(documentv2)
+				// num := strconv.Itoa(numErrors)
+				// if _, err := f.WriteString(env.Request.SourceName + "," + num + "\n"); err != nil {
+				// 	log.Println(err)
+				// }
+			}
+		case "openapi.v3.Document":
+			documentv3 := &openapiv3.Document{}
+			err = proto.Unmarshal(model.Value, documentv3)
+			if err == nil {
+				// Analyze the API document.
+				linter, _ = lint.AIPLinterV3(documentv3)
+			}
+		}
+	}
+
+	if linter != nil {
+		outputName1 := filepath.Join(
+			filepath.Dir(env.Request.SourceName), "linter.json")
+		outputName2 := filepath.Join(
+			filepath.Dir(env.Request.SourceName), "linter.pb")
+		file := &plugins.File{}
+
+		file.Name = outputName1
+		file.Data, err = json.MarshalIndent(linter, "", "  ")
```

Comment:
please remove the outputName1 (and outputName2) variables and put their values inline here.

Reply:
Lines 85 and following should as much as possible look like lines 93 and following - same formatting, line spacing, etc.